spec/lib/hubspot/blog_spec.rb

```diff
@@ -1,116 +1,150 @@
+          expect(result).to be_kind_of(Array)
         end
       end
 
-      describe "can be ordered" do
-        it "created at descending is default" do
-          created_timestamps = blog.posts(created_range_params).map { |post| post['created'] }
-          expect(created_timestamps.sort.reverse).to eq(created_timestamps)
-        end
+      it "includes given parameters in the request" do
+        VCR.use_cassette("blog_posts/filter_blog_posts", record: :none) do
+          blog_id = 123
+          created_gt = timestamp_in_milliseconds(Time.now - 2.months)
+          blog = Hubspot::Blog.new({ "id" => 123 })
+
+          result = blog.posts({ state: "DRAFT" })
 
-        it "by created ascending" do
-          pending
-          created_timestamps = blog.posts({order_by: '+created'}.merge(created_range_params)).map { |post| post['created'] }
-          expect(created_timestamps.sort).to eq(created_timestamps)
+          assert_requested :get, hubspot_api_url("/content/api/v2/blog-posts?content_group_id=#{blog_id}&created__gt=#{created_gt}&hapikey=demo&order_by=-created&state=DRAFT")
+          expect(result).to be_kind_of(Array)
         end
       end
 
-      it "can set a page size" do
-        pending 'Not working'
-        blog.posts({limit: 10}.merge(created_range_params)).length.should be(10)
+      it "raises when given an unknown state" do
+        blog = Hubspot::Blog.new({})
+
+        expect {
+          blog.posts({ state: "unknown" })
+        }.to raise_error(Hubspot::InvalidParams, "State parameter was invalid")
       end
     end
   end
 
   describe Hubspot::BlogPost do
-    cassette "blog_posts"
+    describe "#created_at" do
+      it "returns the created timestamp as a Time" do
+        timestamp = timestamp_in_milliseconds(Time.now)
+        blog_post = Hubspot::BlogPost.new({ "created" => timestamp })
 
-    let(:example_blog_post) do
-      VCR.use_cassette("one_month_blog_posts_filter_state", record: :none) do
-        blog = Hubspot::Blog.new(example_blog_hash)
-        blog.posts(created_range_params).first
+        expect(blog_post.created_at).to eq(Time.at(timestamp/1000))
       end
     end
 
-    it "should have a created_at value specific method" do
-      expect(example_blog_post.created_at).to eq(Time.at(example_blog_post['created'] / 1000))
-    end
+    describe ".find_by_blog_post_id" do
+      it "retrieves a blog post by id" do
+        VCR.use_cassette "blog_posts" do
+          blog_post_id = 422192866
 
-    it "can find by blog_post_id" do
-      blog = Hubspot::BlogPost.find_by_blog_post_id(422192866)
-      expect(blog['id']).to eq(422192866)
+          result = Hubspot::BlogPost.find_by_blog_post_id(blog_post_id)
+
+          assert_requested :get, hubspot_api_url("/content/api/v2/blog-posts/#{blog_post_id}?hapikey=demo")
+          expect(result).to be_a(Hubspot::BlogPost)
```

Comment:
I've seen this now a couple of times and I think I'm a bit confused.

That said, I can't decide what I think what I should ask to change. It seems strange to use VCR in a unit test and maybe it'd be better served to just confirm that something was returned, but on the other hand that request is the basis of how this class works, so it makes sense to me send a request.

I have nothing constructive to add, but wanted to say something.

Reply:
Looking at this further I don't know exactly what I'm asking.

I understand that if you are using a class method that is expected to return an instance of that class, that is a good thing. I also understand that if you are doing a http request you want to test that http request has been made.

I'm having trouble articulating, but the more I look at this and try to rationalize my thoughts the more I think you solved this the right way. 

So please feel free to ignore.